sql: plumb isCopy down to event log

otan opened the pull request:
Previously, we relied on the `txn` being filled to log `COPY`
statements. This is however inaccurate - `txn` may be nil if `COPY` is
the first statement encountered. This meant that the first COPY
encountered would not be logged.

Instead, plumb down `isCopy` and use `timeutil.Now()` if it is a COPY
related statement.

Release justification: logging related critical fix

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/87151)
<!-- Reviewable:end -->

otan committed:
sql: plumb isCopy down to event log

Previously, we relied on the `txn` being filled to log `COPY`
statements. This is however inaccurate - `txn` may be nil if `COPY` is
the first statement encountered. This meant that the first COPY
encountered would not be logged.

Instead, plumb down `isCopy` and use `timeutil.Now()` if it is a COPY
related statement.

Release justification: logging related critical fix

Release note: None

rafiss reviewed (approved):
lgtm

rafiss commented on pkg/sql/conn_executor.go:
```diff
@@ -2457,6 +2457,7 @@ func (ex *connExecutor) execCopyIn(
 			ctx,
 			ex.executorType,
+			true, /* isCopy */
 			int(ex.state.mu.autoRetryCounter),
```
completely unrelated, but i just realized we are accessing `mu` here, but didn't take the lock. maybe it's fine since nothing else can really happen during COPY

otan replied:
mmmm i think we do the same thing in the "normal path"

otan commented:
bors r=rafiss
